spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/TextEncryptorLocator.java

```diff
@@ -0,0 +1,25 @@
+package org.springframework.cloud.config.server.encryption;
+
+import org.springframework.security.crypto.encrypt.TextEncryptor;
+
+public class TextEncryptorLocator {
+
+    private TextEncryptor encryptor;
+
+    public TextEncryptorLocator() {
+    }
+
+    public TextEncryptorLocator(TextEncryptor encryptor) {
+        this.encryptor = encryptor;
+    }
+
+    // temporary solution to support EncryptionController REST API
+    public void setEncryptor(TextEncryptor encryptor) {
+        this.encryptor = encryptor;
+    }
+
+    public TextEncryptor get() {
```

Comment:
In the future, TextEncryptorLocator could be used to find an ecryptor dedicated for given application profile (hopefully with some new finders methods). 
In this PR it is not very usefeul yet, but at least it extracts a state (TextEncryptor) from other controllers. 
Currently TextEncryptor is injected and at the same time there is a method to update key. 
This scenario is improved by the abstraction, as updating a key will update a property of singleton bean instead of recreating a property in a EncryptionController. You don't need to depend on EncryptionController if you want to have access to encryptor, just inject the TextEncryptorLocator.